diesel_derives/src/for_each_tuple.rs

```diff
@@ -0,0 +1,977 @@
+            };
+        }
+
+        syn::visit_mut::visit_expr_block_mut(self, expr);
+        timed_block!(time, "visit_expr_block_mut", self.timing);
+    }
+
+    fn visit_impl_item_const_mut(&mut self, const_impl: &mut syn::ImplItemConst) {
+        let time = start_timed_block!("visit_impl_item_const_mut", self.timing);
+        if let syn::Expr::Binary(syn::ExprBinary {
+            left, right, op, ..
+        }) = const_impl.expr.clone()
+        {
+            if let syn::Expr::Path(syn::ExprPath {
+                qself: None, path, ..
+            }) = *left
+            {
+                let mut path_iter = path.segments.into_iter();
+                if let Some(first) = path_iter.next() {
+                    let remaining_segments = path_iter.collect::<Vec<_>>();
+                    if let Some(param) = self.repeated_types.iter().find(|t| t.ident == first.ident)
+                    {
+                        let expr = self.index_range.clone().map(move |idx| {
+                            let ident =
+                                Ident::new(&format!("{}_{}", param.ident, idx), first.span());
+                            let segments = std::iter::once(syn::PathSegment {
+                                ident: ident,
+                                arguments: syn::PathArguments::None,
+                            })
+                            .chain(remaining_segments.clone())
+                            .collect();
+                            syn::Path {
+                                leading_colon: None,
+                                segments,
+                            }
+                        });
+                        const_impl.expr = parse_quote!(#(#expr #op)* #right);
+                    }
+                }
+            }
+        }
+
+        syn::visit_mut::visit_impl_item_const_mut(self, const_impl);
+        timed_block!(time, "visit_impl_item_const_mut", self.timing);
+    }
+
+    fn visit_expr_mut(&mut self, expr: &mut syn::Expr) {
+        let time = start_timed_block!("visit_expr_mut", self.timing);
+        if let syn::Expr::Macro(macro_expr) = expr {
+            if macro_expr.mac.path.segments.last().unwrap().ident == "tuple_size" {
+                let span = expr.span();
+                let tuple_size = self.index_range.end;
+                *expr = syn::Expr::Lit(syn::ExprLit {
+                    attrs: Vec::new(),
+                    lit: syn::Lit::Int(syn::LitInt::new(&tuple_size.to_string(), span)),
+                });
+            }
+        }
+
+        syn::visit_mut::visit_expr_mut(self, expr);
+        timed_block!(time, "visit_expr_mut", self.timing);
+    }
+}
+
+struct ExprReplacer<'a> {
+    idx: u32,
+    repeated_types: &'a [TypeParam],
+}
+
+impl<'a> syn::visit_mut::VisitMut for ExprReplacer<'a> {
+    fn visit_expr_field_mut(&mut self, i: &mut syn::ExprField) {
+        if let syn::Member::Named(n) = i.member.clone() {
+            if n == "idx" {
```

Comment:
Even though this is an internal macro, I don't like the fact that we override `.idx`. Can we do `._idx` or something to make sure it is different to a field?